app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/data/mapper/MessageActionMapper.kt

```diff
@@ -14,7 +14,7 @@ class MessageActionMapper @Inject constructor() {
     fun map(action: String?): Result<MessagesAction> {
         return runCatching {
```

Comment:
there is another error being thrown in the else case. I would propose changing the return type of this method to `MessageAction?` instead of result and update to `runCatching { // exisiting logic }.getOrNull()`. No need for NoAction imo as we can represent it with the null case

Reply:
Changed it accordingly